File: spring-microservices/course-service/src/main/java/com/hoangphihiep/dto/response/QuizResultResponse.java
package com.hoangphihiep.dto.response;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.Date;
import java.util.List;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class QuizResultResponse {

    private Integer id;

    private String studentId;

    private String studentName;

    private String email;

    private Integer quizId;

    private String quizTitle;

    private String quizDescription;

    private Double score;

    private Double totalScore;

    private Double percentage;

    private Boolean isPassed;

    private Integer attemptNumber;

    private Date startedAt;

    private Date submittedAt;

    private Integer timeSpent; // in seconds

    private List<QuizAnswerResponse> answers;
}

File: react-type-vite/src/services/api/response/quizResultResponse.ts
import type { QuizAnswerResponse } from "./quizAnswerResponse"

export interface QuizResultResponse {
  id: number
  studentId: string
  studentName: string
  email: string
  quizId: number
  quizTitle: string
  quizDescription?: string
  score: number
  totalScore: number
  percentage: number
  isPassed: boolean
  attemptNumber: number
  startedAt: string
  submittedAt: string
  timeSpent: number // in seconds
  answers: QuizAnswerResponse[]
}